core/chains/evm/log/registrations.go

```diff
@@ -196,80 +240,83 @@ func filtersContainValues(topicValues []common.Hash, filters [][]Topic) bool {
 	return true
 }
 
-func newHandlers(evmChainID big.Int) *handlers {
-	return &handlers{
-		handlersByAddr: make(map[common.Address]map[common.Hash]map[Listener]*listenerMetadata),
-		evmChainID:     evmChainID,
+func newHandlers(lggr logger.Logger, evmChainID big.Int) *handler {
+	return &handler{
+		lookupSubscribers: make(map[common.Address]map[common.Hash]map[*subscriber][][]Topic),
+		evmChainID:        evmChainID,
+		logger:            lggr,
 	}
 }
 
-func (r *handlers) addSubscriber(sub *subscriber) (needsResubscribe bool) {
+func (r *handler) addSubscriber(sub *subscriber) (needsResubscribe bool) {
 	addr := sub.opts.Contract
 
 	if sub.opts.MinIncomingConfirmations <= 0 {
-		sub.opts.MinIncomingConfirmations = 1
+		r.logger.Panicw(fmt.Sprintf("LogBroadcaster requires that MinIncomingConfirmations must be at least 1 (got %v). Logs must have been confirmed in at least 1 block, it does not support reading logs from the mempool before they have been mined.", sub.opts.MinIncomingConfirmations), "addr", sub.opts.Contract.Hex(), "jobID", sub.listener.JobID())
 	}
 
-	if _, exists := r.handlersByAddr[addr]; !exists {
-		r.handlersByAddr[addr] = make(map[common.Hash]map[Listener]*listenerMetadata)
+	if _, exists := r.lookupSubscribers[addr]; !exists {
+		r.lookupSubscribers[addr] = make(map[common.Hash]map[*subscriber][][]Topic)
 	}
 
 	for topic, topicValueFilters := range sub.opts.LogsWithTopics {
-		if _, exists := r.handlersByAddr[addr][topic]; !exists {
-			r.handlersByAddr[addr][topic] = make(map[Listener]*listenerMetadata)
+		if _, exists := r.lookupSubscribers[addr][topic]; !exists {
+			r.lookupSubscribers[addr][topic] = make(map[*subscriber][][]Topic)
```

Comment:
Defining a type like `type subscribers map[*subscriber][][]Topic` might simplify some things :shrug: 
```suggestion
			r.lookupSubscribers[addr][topic] = make(subscribers)
```